solidity/orders/src/AdvancedOrder.sol

```diff
@@ -0,0 +1,152 @@
+
+        uint256[] memory predictedPrices = abi.decode(future.futureResponse.result.output, (uint256[]));
+        GetPriceResponse memory priceResponse =
+            SLINKY_PRECOMPILE.getPrice(orderData.oraclePricePair.base, orderData.oraclePricePair.quote);
+
+        uint256 predictedPrice = _getPriceInQuote(predictedPrices[0], predictedPrices[1], PRICE_PREDICTION_DECIMALS);
+        (uint256 oracleNormalized, uint256 predictedNormalized) =
+            _normalizePrices(
+                priceResponse.price.price,
+                predictedPrice,
+                priceResponse.decimals,
+                PRICE_PREDICTION_DECIMALS
+            );
+
+        return _checkCondition(oracleNormalized, predictedNormalized);
+    }
+
+    function execute(
+        uint256 nonce,
+        uint256 gas,
+        uint256,
+        uint256 maxPriorityFeePerGas,
+        uint256 maxFeePerGas
+    )
+        external
+        override
+        returns (bool, bytes32)
+    {
+        if (msg.sender != _scheduler) revert Unauthorized();
+        if (_executed) revert ExecutedError();
+        if (!canExecute()) revert ConditionNotMet();
+
+        bytes[] memory emptyAccessList = new bytes[](0);
+        (bytes memory unsignedTx, bytes32 txHash) = this.encodeUnsignedEIP1559(
+            nonce, gas, maxPriorityFeePerGas, maxFeePerGas, emptyAccessList, commonExecutionData.creatorDefinedTxFields
+        );
+
+        _unsignedTx = unsignedTx;
+        _executed = this.createSignRequest(commonExecutionData.signRequestData, abi.encodePacked(txHash), _coins);
+
+        if (_executed) emit Executed();
+
+        // origin always creator of sign request
+        // solhint-disable-next-line
+        REGISTRY.addTransaction(tx.origin, txHash);
+        return (_executed, txHash);
+    }
+
+    function isExecuted() public view override returns (bool) {
+        return _executed;
+    }
+
+    function getTx() external view returns (bytes memory) {
+        if (!_executed) revert ExecutedError();
+        return _unsignedTx;
+    }
+
+    function executionData() external view returns (ExecutionData memory data) {
+        data = this.buildExecutionData(commonExecutionData.creatorDefinedTxFields);
+    }
+
+    function _normalizePrices(
+        uint256 price1, uint256 price2, uint256 decimals1, uint256 decimals2
+    ) internal pure returns (uint256 normalizedPrice1, uint256 normalizedPrice2) {
+        uint256 maxDecimals = decimals1 > decimals2 ? decimals1 : decimals2;
+        normalizedPrice1 = price1 * (10**(maxDecimals - decimals1));
+        normalizedPrice2 = price2 * (10**(maxDecimals - decimals2));
+    }
+
+    function _getPriceInQuote(
+        uint256 priceA, uint256 priceB, uint256 decimals
+    ) internal pure returns (uint256) {
+        return (priceA * 10**decimals) / priceB;
+    }
+
+    function _checkCondition(uint256 oraclePrice, uint256 predictedPrice) internal view returns (bool) {
```

Comment:
can we rename this function to sound less generic?

Reply:
`checkPriceCondition`?